api/client.go

```diff
@@ -43,18 +43,21 @@ type WrappingLookupFunc func(operation, path string) string
 
 // Config is used to configure the creation of the client.
 type Config struct {
+	modifyLock sync.RWMutex
```

Comment:
Nitpick here, but can we go for embedding sync.Mutex so that we call `c.Lock()` vs `c.modifyLock.Lock()`?

Reply:
I don't want to allow API consumers to toggle the lock...doing it this way makes it unexpired.

Reply:
Oops, unexported.

Reply:
Add that makes sense, nvm then!